File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
const colorMap:string[] = [
  'black',
  'brown',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'violet',
  'grey',
  'white',
];

export function decodedValue(prop : string[]) : number {
  const indexArray:number[] = [];
  for (let i=0; i<prop.length && i<2; ++i) {
    indexArray.push(colorMap.indexOf(prop[i]));
    if (indexArray[i] === -1 || Number.isNaN(indexArray[i]) || indexArray[i] === Infinity || indexArray[i] === -Infinity)
    {
      throw new Error(`Unsupported color at index: ${i}`)
    }
  }
  if (indexArray.length !== 2) {
    throw new Error("Need at least 2 colors");
  }
  return indexArray.reduce((acc:number, cur:number) => {
    return acc * 10 + cur;
  }, 0)
}

File: solutions/go/resistor-color-duo/1/resistor_color_duo.go
package resistorcolorduo

import "slices"

var Colors = []string{
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
}

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	return ColorCode(colors[0]) * 10 + ColorCode(colors[1])
}

func ColorCode(color string) int {
    return slices.Index(Colors, color)
}